src/addrdb.cpp

```diff
@@ -61,25 +62,29 @@ bool SerializeFileDB(const std::string& prefix, const fs::path& path, const Data
     FILE *file = fsbridge::fopen(pathTmp, "wb");
     AutoFile fileout{file};
     if (fileout.IsNull()) {
-        fileout.fclose();
         remove(pathTmp);
         LogError("%s: Failed to open file %s\n", __func__, fs::PathToString(pathTmp));
         return false;
     }
 
     // Serialize
     if (!SerializeDB(fileout, data)) {
-        fileout.fclose();
+        (void)fileout.fclose();
         remove(pathTmp);
         return false;
     }
     if (!fileout.Commit()) {
-        fileout.fclose();
+        (void)fileout.fclose();
         remove(pathTmp);
         LogError("%s: Failed to flush file %s\n", __func__, fs::PathToString(pathTmp));
         return false;
     }
-    fileout.fclose();
+    if (fileout.fclose() != 0) {
+        const int errno_save{errno};
+        remove(pathTmp);
+        LogError("Failed to close file %s: %s", fs::PathToString(pathTmp), SysErrorString(errno_save));
```

Comment:
In other cased you've added `after commit` to clarify when closing failed.

Also previous errors we've prefixed with `__func__`:
```suggestion
        LogError("s: Failed to close file %s after commit: %s", __func__, fs::PathToString(pathTmp), SysErrorString(errno_save));
```

Looking at the other examples you've changed, I assume this was deliberate - I'd still prefer unifying them, but I don't mind tackling it in a separate PR either.

Reply:
> __func__ is redundant

Yes, would be great to unify the affected messages - either by adding or removing in every touched method. But we can of course do it in a separate PR as well.